src/components/Card.jsx

```diff
@@ -1,56 +1,124 @@
         >
-          <div className="md:flex items-center  ">
-            <img
-              className="w-full object-cover h-full md:w-48"
-              loading="lazy"
-              src={exercise.image}
-              alt={exercise.exercise}
-            />
-            <div className="p-8">
-              <div className="uppercase tracking-wide text-sm text-indigo-800 font-semibold">
-                {exercise.exercise}
-              </div>
+          <option value="All">All</option>
+          {uniqueTags.map((tag, index) => (
+            <option key={index} value={tag}>
+              {tag}
+            </option>
+          ))}
+        </select>
+      </div>
+
+      <div className="grid grid-cols-1 p-3 lg:grid-cols-2">
+        <AnimatePresence key={selectedTag}>
+          {sortedExercises.map((exercise, index) => (
+            <motion.div
+              key={index}
+              className="max-w-md mx-auto my-4 overflow-hidden bg-white border-2 shadow-md rounded-xl md:max-w-2xl hover:border-purple-500"
+              initial="initial"
+              animate="animate"
+              exit="exit"
+              variants={cardVariants}
+            >
+              <div className="items-center md:flex ">
+                <img
+                  className="object-cover w-full h-full md:w-48"
+                  loading="lazy"
+                  src={exercise.image}
+                  alt={exercise.exercise}
+                />
+                <div className="p-8">
+                  <div className="text-sm font-semibold tracking-wide text-indigo-800 uppercase">
+                    {exercise.exercise}
+                  </div>
+
+                  <ul className="p-2 text-gray-500 list-disc">
+                    {exercise.instructions.map((instruction, index) => (
+                      <li key={index}>{instruction}</li>
+                    ))}
+                  </ul>
 
-              <ul className="p-2 text-gray-500 list-disc">
-                {exercise.instructions.map((instruction, index) => (
-                  <li key={index}>{instruction}</li>
-                ))}
-              </ul>
-              <div className="mt-6 gap-5 flex items-center">
-                <a
-                  href={exercise.videoLink}
-                  target="_blank"
-                  rel="noreferrer"
-                  className="text-indigo-800 hover:text-indigo-600 font-semibold"
-                >
-                  Watch video
-                </a>
-                <span>
-                  Added by :{" "}
-                  <a
-                    href={`https://github.com/${exercise["gh-name"]}`}
-                    target="_blank"
-                    rel="noreferrer"
-                    className="text-indigo-800 hover:text-indigo-600 font-semibold"
-                  >
-                    {exercise["gh-name"]}
-                  </a>
-                </span>
+                  <div className="flex flex-wrap mt-2">
```

Comment:
If someone has not added any tags. Would this add an extra horizontal space? Is it okay from a UI perspective.